django/db/models/query.py

```diff
@@ -870,18 +865,14 @@ def annotate(self, *args, **kwargs):
         """
         annotations = OrderedDict()  # To preserve ordering of args
         for arg in args:
-            # The default_alias property may raise a TypeError, so we use
-            # a try/except construct rather than hasattr in order to remain
-            # consistent between PY2 and PY3 (hasattr would swallow
-            # the TypeError on PY2).
             try:
+                annotations[arg.default_alias] = arg
                 if arg.default_alias in kwargs:
                     raise ValueError("The named annotation '%s' conflicts with the "
                                      "default name for another annotation."
                                      % arg.default_alias)
             except (AttributeError, TypeError):
```

Comment:
From a quick check on SQLite, `AttributeError` can be removed here without any test failures. Might be missing test coverage.